components/forms/TransactionSigning.tsx

```diff
@@ -44,12 +49,23 @@ const TransactionSigning = (props: Props) => {
         sign: { preferNoSetFee: true, preferNoSetMemo: true, disableBalanceCheck: true },
       };
       const tempWalletAccount = await window.keplr.getKey(state.chain.chainId);
-      console.log(tempWalletAccount);
-      const tempHasSigned = props.signatures.some(
-        (sig) => sig.address === tempWalletAccount.bech32Address,
-      );
       setWalletAccount(tempWalletAccount);
-      setHasSigned(tempHasSigned);
+
+      const pubkey = toBase64(tempWalletAccount.pubKey);
+      const isMember = memberPubkeys.includes(pubkey);
+      const hasSigned = isMember
+        ? props.signatures.some((sig) => sig.address === tempWalletAccount.bech32Address)
+        : false;
+      if (hasSigned) {
+        setSigning("signed");
+      }
+      if (isMember && !hasSigned) {
+        setSigning("not_signed");
+      }
+      if (!isMember) {
+        setSigning("not_a_member");
+      }
```

Comment:
You could write those something like this to ensure `setSigning` is only called one time. But I don't mind much as the code seems to be correct.

```suggestion     
      if (!isMember) {
        setSigning("not_a_member");
      } else if {
           if (hasSigned) {
                setSigning("signed");
          } else {
                setSigning("not_signed");
          }
      }
```